plugins/catalog/src/alpha/filter/parseFilterExpression.ts

```diff
@@ -73,9 +78,12 @@ export function parseFilterExpression(expression: string): {
       return [];
     }
 
-    const matcher = factory(part.parameters, e =>
-      expressionParseErrors.push(e),
+    const matcher = factory(
+      part.parameters,
+      e => expressionParseErrors.push(e),
+      negation,
```

Comment:
No need to pass this down. If it's per part, then only this bit of the code needs to know how to negate the return value of the factory output

Reply:
Trying my to wrap my head around the flow of this part with everything you've mentioned - 

Are you thinking the part coming through looks like:

`{key: 'not:type', parameter:'service'} `

And the factory determines to negate the response? Or that the flat map returns something different when it's negated and then that's handled by the `filterFn?

Reply:
I'm saying that the code in here, would strip out the `not:` and the factory would never even be aware that it existed. Our code would just memorize that there was a negation, so that it could return the inverse of what the factory returned.